src/main/java/uk/gov/hmcts/darts/audio/controller/AudioController.java

```diff
@@ -77,12 +80,62 @@ public ResponseEntity<Resource> preview(Integer mediaId) {
         return new ResponseEntity<>(new InputStreamResource(audioMediaFile), HttpStatus.OK);
     }
 
+    @SneakyThrows
+    @Override
+    @SecurityRequirement(name = SECURITY_SCHEMES_BEARER_AUTH)
+    @Authorisation(contextId = MEDIA_ID,
+        securityRoles = {JUDGE, REQUESTER, APPROVER, TRANSCRIBER, LANGUAGE_SHOP_USER, RCJ_APPEALS})
+    public ResponseEntity<byte[]> preview2(Integer mediaId, @RequestHeader(value = "Range", required = false) String httpRangeList) {
+        InputStream audioMediaFile = audioService.preview(mediaId);
+
+        if (httpRangeList == null) {
+            byte[] bytes = IOUtils.toByteArray(audioMediaFile);
+            return ResponseEntity.status(HttpStatus.OK)
+                .header("Content-Type", "audio/mpeg")
+                .header("Content-Length", String.valueOf(bytes.length))
+                .body(bytes);
+        } else {
+            byte[] bytes = IOUtils.toByteArray(audioMediaFile);
+            long fileSize = bytes.length;
+            String[] ranges = httpRangeList.split("-");
+            long rangeStart = Long.parseLong(ranges[0].substring(6));
+            long rangeEnd;
+            if (ranges.length > 1) {
+                rangeEnd = Long.parseLong(ranges[1]);
+            } else {
+                rangeEnd = fileSize - 1;
+            }
+            if (fileSize < rangeEnd) {
+                rangeEnd = fileSize - 1;
+            }
+            String contentLength = String.valueOf((rangeEnd - rangeStart) + 1);
+            String contentRange = "bytes" + " " + rangeStart + "-" + rangeEnd + "/" + fileSize;
+            return ResponseEntity.status(HttpStatus.OK)
```

Comment:
@davet1985 Not sure it will help, but wondering if the status here should be PARTIAL_CONTENT (206)?
It might do something in combination with the special Content-Range header perhaps...

Reply:
@richardbull yes it should be, thank you for pointing that out 🤦